### PR TITLE
Normalize val in BitVector(val_str, base) 

### PR DESCRIPTION
**Pull request by alex-ozdemir**
Fixes cpp API's mkBitVector(val_str, base) constructor.

### CODE REVIEW

File: src/api/cpp/cvc5.cpp [hunk 1]
```diff
@@ -4862,6 +4864,8 @@ Term Solver::mkBVFromStrHelper(uint32_t size,
         << "Overflow in bitvector construction (specified bitvector size "
         << size << " too small to hold value " << s << ")";
   }
+  // Doesn't happen automatically.
+  val = val.modByPow2(size);
```
Comment (aniemetz) [inline src/api/cpp/cvc5.cpp:4868]: This is done in the `BitVector` constructor, actually: https://github.com/cvc5/cvc5/blob/master/src/util/bitvector.h#L33

**Review by aniemetz — CHANGES_REQUESTED** [review]
Can you also add a test to the solver_black tests?
https://github.com/cvc5/cvc5/blob/master/test/unit/api/solver_black.cpp#L307

File: src/api/cpp/cvc5.cpp (hunk 1 above)
Comment (alex-ozdemir) [reply src/api/cpp/cvc5.cpp]: Good point. Removed.

**Comment by alex-ozdemir** [comment]
Ah---I didn't see that test file.

I've added tests for the current pattern of errors. You can amend them in your refactor, if we throw errors more liberally then.

**Review by aniemetz — APPROVED** [review]

**Pull request merged**